src/recon_test/test_Transformations.cxx

```diff
@@ -0,0 +1,223 @@
+  void run_test_motion_fwrd_and_back();
+
+public:
+  /// Set inputs
+  void set_inputs(string fwrd_output,
+                  string back_output,
+                  string to_transform,
+                  string disp_field_x,
+                  string disp_field_y,
+                  string disp_field_z);
+
+  /// Set inputs
+  void set_inputs(string fwrd_output,
+                  string back_output,
+                  string to_transform,
+                  string disp_field_multi_component);
+};
+
+TransformationTests::
+TransformationTests()
+{
+    _bspline_order = 3;
+}
+
+void
+TransformationTests::
+set_inputs(string fwrd_output, string back_output, string to_transform, string disp_field_x, string disp_field_y, string disp_field_z)
+{
+    _disp_field_multi_component = "";
+    _disp_field_x = disp_field_x;
+    _disp_field_y = disp_field_y;
+    _disp_field_z = disp_field_z;
+    _to_transform = to_transform;
+    _fwrd_output  = fwrd_output;
+    _back_output  = back_output;
+}
+
+void
+TransformationTests::
+set_inputs(string fwrd_output, string back_output, string to_transform, string disp_field_multi_component)
+{
+    _disp_field_multi_component = disp_field_multi_component;
+    _disp_field_x = "";
+    _disp_field_y = "";
+    _disp_field_z = "";
+    _to_transform = to_transform;
+    _fwrd_output  = fwrd_output;
+    _back_output  = back_output;
+}
+
+void
+TransformationTests::
+run_test_motion_fwrd_and_back()
+{
+    ITKOutputFileFormat output_file_format;
+    output_file_format.default_extension = ".nii";
+
+    // Open image
+    shared_ptr<DiscretisedDensity<3,float> > input;
+    std::cerr << "\nabout to read the image to transform...\n";
+    input.reset( DiscretisedDensity<3,float>::read_from_file(_to_transform));
+
+    // Create non rigid transformations
+    shared_ptr<NonRigidObjectTransformationUsingBSplines<3,float> > fwrd_non_rigid;
+    if (_disp_field_multi_component != "") {
+        fwrd_non_rigid.reset(new NonRigidObjectTransformationUsingBSplines<3,float>(_disp_field_multi_component,_bspline_order));
+    }
+    else if (_disp_field_x != "" && _disp_field_y != "" && _disp_field_z != "")
+        fwrd_non_rigid.reset(new NonRigidObjectTransformationUsingBSplines<3,float>(_disp_field_x,_disp_field_y,_disp_field_z,_bspline_order));
+
+    // Create image processors
+    shared_ptr<Transform3DObjectImageProcessor<float> > fwrd_transform, back_transform;
+    fwrd_transform.reset( new Transform3DObjectImageProcessor<float>(fwrd_non_rigid) );
+    back_transform.reset( new Transform3DObjectImageProcessor<float>(*fwrd_transform) );
+    back_transform->set_do_transpose(!fwrd_transform->get_do_transpose());
+
+    // Start the timer
+    HighResWallClockTimer t;
+    t.reset();
+    t.start();
+
+    shared_ptr<DiscretisedDensity<3,float> > forward;
+    forward.reset(input->clone());
+    fwrd_transform->apply(*forward);
+    output_file_format.write_to_file(_fwrd_output, *forward);
```

Comment:
not really a test yet

Reply:
I added a small multicomponent nifti image (couple of kB). I'll also split that into its x-, y- and z-components and then check that the forward and transpose transformations are the same regardless of whether multicomponent or single components are used.

I'll do this soon, so no need to merge before then.